### PR TITLE
Add config options for power consumption of individual network components

### LINKED ISSUE #856: More Configurability of Network Energy Consumption
#### Issue type:

- :heavy_plus_sign: Feature request

____

#### Short description:

I'm planning on assembling a personal modpack in which ID network power consumption is enabled, but I noticed that the relative energy consumption values for each network part are hard-coded. For example, I'd like to make storage terminals rather energy-hungry, but I can't do so without making variable displays require an equal amount of energy, and variable stores would require twice as much energy. It would be nice if I could manually set the relative energy consumption values for each part in the configs.

### PR DESCRIPTION
**Pull request by Trainerredstone7**
Resolves #856.

### CODE REVIEW

File: src/main/java/org/cyclops/integrateddynamics/part/PartTypePanelDisplay.java [hunk 1]
```diff
@@ -32,7 +33,7 @@ protected Block createBlock(BlockConfig blockConfig) {
 
     @Override
     public int getConsumptionRate(State state) {
-        return state.hasVariable() ? 2 : 1;
+        return GeneralConfig.panelDisplayBaseConsumption;
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/part/PartTypePanelDisplay.java]: Could you add `panelDisplayBaseConsumptionEnabled` and `panelDisplayBaseConsumptionDisabled` here instead?

**Review by rubensworks — CHANGES_REQUESTED** [review]
Looks good!

Having separate fields for every part is a bit unfortunate, but it's the best we can do with the current config system I guess. I should be able to improve this in 1.15 though.

File: src/main/java/org/cyclops/integrateddynamics/part/PartTypePanelDisplay.java (hunk 1 above)
Comment (Trainerredstone7) [reply src/main/java/org/cyclops/integrateddynamics/part/PartTypePanelDisplay.java]: I've split the display panel config option into separate options for enabled and disabled states.

**Comment by Trainerredstone7** [comment]
I'll work on making similar pull requests to add power consumption config options to Integrated Tunnels, Crafting, and Terminals.

**Comment by rubensworks** [comment]
Great, thanks!

**Pull request merged**